File: test/corpus/advanced/02-go-api-full-compatibility/case.go
package main

import (
	"fmt"

	"github.com/muesli/termenv"
)

func main() {
	fmt.Println("--- Go API Full Compatibility Test ---")
	
	// Test global convenience functions (PascalCase)
	fmt.Printf("ColorProfile: %d\n", int(termenv.ColorProfile()))
	fmt.Printf("EnvColorProfile: %d\n", int(termenv.EnvColorProfile()))
	fmt.Printf("EnvNoColor: %t\n", termenv.EnvNoColor())
	fmt.Printf("HasDarkBackground: %t\n", termenv.HasDarkBackground())
	
	// Test color creation functions
	rgbColor := termenv.RGBColor("#FF0000")
	ansiColor := termenv.ANSIColor(1)
	ansi256Color := termenv.ANSI256Color(196)
	
	fmt.Printf("RGBColor sequence: %s\n", rgbColor.Sequence(false))
	fmt.Printf("ANSIColor: %s\n", ansiColor.String())
	fmt.Printf("ANSI256Color: %s\n", ansi256Color.String())
	
	// Test String function (global convenience)
	globalString := termenv.String("Global String Test")
	fmt.Printf("Global String (plain): %s\n", globalString)
	
	globalStyled := termenv.String("Global Styled").Bold().Foreground(rgbColor)
	fmt.Printf("Global String (styled): %s\n", globalStyled)
	
	// Test NewOutput function
	out := termenv.NewOutput(nil, termenv.WithProfile(termenv.TrueColor))
	
	// Test Output methods
	outputString := out.String("Output String Test")
	fmt.Printf("Output String (plain): %s\n", outputString)
	
	outputStyled := out.String("Output Styled").Bold().Foreground(out.Color("#00FF00"))
	fmt.Printf("Output String (styled): %s\n", outputStyled)
	
	// Test String() method on Style
	style := out.String("Style String Method").Bold().Foreground(rgbColor)
	fmt.Printf("Style.String(): %s\n", style.String())
	
	// Test all style methods (PascalCase methods)
	allStyles := out.String("All Styles").
		Bold().
		Faint().
		Italic().
		Underline().
		Overline().
		Blink().
		Reverse().
		CrossOut().
		Foreground(rgbColor).
		Background(ansi256Color)
	
	fmt.Printf("All Styles: %s\n", allStyles)
	
	// Test color factory through Output.Color
	outputRgbColor := out.Color("#FF6600")
	outputColorStyled := out.String("Output Color").Foreground(outputRgbColor)
	fmt.Printf("Output Color: %s\n", outputColorStyled)
	
	// Test WithProfile, WithTTY, WithEnvironment, WithColorCache, WithUnsafe
	configuredOut := termenv.NewOutput(nil,
		termenv.WithProfile(termenv.ANSI256),
		termenv.WithTTY(true),
		termenv.WithColorCache(true),
		termenv.WithUnsafe())
	
	configuredStyled := configuredOut.String("Configured Output").Bold().Foreground(termenv.ANSI256Color(46))
	fmt.Printf("Configured Output: %s\n", configuredStyled)
}

File: test/corpus/advanced/02-go-api-full-compatibility/case.ts
// biome-ignore lint/suspicious/noShadowRestrictedNames: Go API compatibility requires this name
import {
  ANSI256,
  ANSI256Color,
  ANSIColor,
  ColorProfile,
  EnvColorProfile,
  EnvNoColor,
  HasDarkBackground,
  NewOutput,
  RGBColor,
  String,
  TrueColor,
  WithColorCache,
  WithProfile,
  WithTTY,
  WithUnsafe,
} from '../../../../src/go-style.js';

console.log('--- Go API Full Compatibility Test ---');

// Test global convenience functions (PascalCase)
console.log(`ColorProfile: ${ColorProfile()}`);
console.log(`EnvColorProfile: ${EnvColorProfile()}`);
console.log(`EnvNoColor: ${EnvNoColor()}`);
console.log(`HasDarkBackground: ${HasDarkBackground()}`);

// Test color creation functions
const rgbColor = RGBColor('#FF0000');
const ansiColor = ANSIColor(1);
const ansi256Color = ANSI256Color(196);

console.log(`RGBColor sequence: ${rgbColor.sequence(false)}`);
console.log(`ANSIColor: ${ansiColor.toString()}`);
console.log(`ANSI256Color: ${ansi256Color.toString()}`);

// Test String function (global convenience)
const globalString = String('Global String Test');
console.log(`Global String (plain): ${globalString.toString()}`);

const globalStyled = String('Global Styled').bold().foreground(rgbColor);
console.log(`Global String (styled): ${globalStyled.toString()}`);

// Test NewOutput function
const out = NewOutput(process.stdout, WithProfile(TrueColor));

// Test Output methods
const outputString = out.string('Output String Test');
console.log(`Output String (plain): ${outputString.toString()}`);

const outputStyled = out.string('Output Styled').bold().foreground(out.color('#00FF00')!);
console.log(`Output String (styled): ${outputStyled.toString()}`);

// Test String() method on Style
const style = out.string('Style String Method').bold().foreground(rgbColor);
console.log(`Style.String(): ${style.toString()}`);

// Test all style methods
const allStyles = out
  .string('All Styles')
  .bold()
  .faint()
  .italic()
  .underline()
  .overline()
  .blink()
  .reverse()
  .crossOut()
  .foreground(rgbColor)
  .background(ansi256Color);

console.log(`All Styles: ${allStyles.toString()}`);

// Test color factory through Output.color
const outputRgbColor = out.color('#FF6600')!;
const outputColorStyled = out.string('Output Color').foreground(outputRgbColor);
console.log(`Output Color: ${outputColorStyled.toString()}`);

// Test WithProfile, WithTTY, WithColorCache, WithUnsafe
const configuredOut = NewOutput(
  process.stdout,
  WithProfile(ANSI256),
  WithTTY(true),
  WithColorCache(true),
  WithUnsafe()
);

const configuredStyled = configuredOut
  .string('Configured Output')
  .bold()
  .foreground(ANSI256Color(46));
console.log(`Configured Output: ${configuredStyled.toString()}`);
